### ISSUE TITLE
React 19 support

### ISSUE THREAD

**Issue by kossnocorp** [vitest-dev/vitest-browser-react#8]
Hey! Thank you so much for this helpful library 🙏

There's a small problem, though. After upgrading to React 19, released a few days ago, it started breaking as it locks on React 18. I tested it on my project with React 19 by copying the source of `vitest-browser-react` code into my repo, as there's no need for any changes. It just works. I think you can safely expand the dependency version to include 19 and ship it as a new version.

**Comment by MatanBobi** [comment]
I agree with @sheremet-va here, we have `"react": ">18.0.0" which allows these versions ([taken from semver.npmjs.com](https://semver.npmjs.com/)):
![image](https://github.com/user-attachments/assets/7841960d-5b3b-46c7-b6dd-f5a6a2b862f0)

**Comment by xeger** [comment]
Confirming that I can use this pkg with React 19; see [xeger/locatorium#1](https://github.com/xeger/locatorium/pull/1) for proof.

On my initial attempt, I experienced an odd error about `react-dom/client` lacking a default export, or not exporting something. I created the project with React 18, which fixed the issue, and later I upgraded to React 19; the issue did not occur.

I've encountered this error intermittently in the past, not always with React 19. I haven't bothered to investigate, as I'm not sure whether it's a legitimate issue with a package or some issue with transpiler/bundler configuration.
